### PR TITLE
Adds @inaccessible directive

### PR DESCRIPTION
**Pull request by michaelstaib**

### CODE REVIEW

File: spec/Section 2 -- Source Schema.md [hunk 1]
```diff
@@ -211,6 +211,86 @@ resolve entities and which source schemas merely contribute data to entities.
...
+# Composite Schema
+
+type Product {
+  id: ID!
+  internalNote: String
+  price: Float!
+}
+```
+
+The above example removes the field `internalProductById` and the type `Product`
```
Comment (glen-84) [inline spec/Section 2 -- Source Schema.md:273]:
```suggestion
The example below removes the field `internalProductById` and the type `Product`
```

File: spec/Section 2 -- Source Schema.md [hunk 2]
```diff
@@ -211,6 +211,86 @@ resolve entities and which source schemas merely contribute data to entities.
...
+from the the composite schema. However, type system members marked as
+inaccessible can still be used within the composite execution schema to fulfill
```
Comment (glen-84) [inline spec/Section 2 -- Source Schema.md:275]:
```suggestion
inaccessible can still be used within the composite execution schema to fulfil
```

**Review by glen-84 — CHANGES_REQUESTED** [review]

**Comment by glen-84** [comment]
Apologies Michael, `fulfill` is correct for American English. 🤦‍♂️

File: spec/Section 2 -- Source Schema.md [hunk 3]
```diff
@@ -211,6 +211,85 @@ resolve entities and which source schemas merely contribute data to entities.
...
+
+```graphql
+directive @inaccessible on OBJECT | FIELD_DEFINITION
+```
+
+The `@inaccessible` directive is used to prevent specific objects or fields from
+being accessible through the composite schema, even if they are accessible in
+the underlying source schemas.
+
+This directive is particularly useful for restricting access to fields or
```
Comment (michaelstaib) [inline spec/Section 2 -- Source Schema.md:224]:
```suggestion
This directive is useful for restricting access to fields or
```

File: spec/Section 2 -- Source Schema.md [hunk 4]
```diff
@@ -211,6 +211,85 @@ resolve entities and which source schemas merely contribute data to entities.
...
+
+type Query {
+  productById(id: ID!): Product
+  productBySku(sku: String!): Product @inaccessible
+}
+```
+
+The above example declares the key field `sku` to be inaccessible from
+the composite schema. However, type system members marked as inaccessible can
+still be used within the composite execution schema to fulfill requirements or
```
Comment (michaelstaib) [inline spec/Section 2 -- Source Schema.md:243]:
```suggestion
still be used within the _execution schema_ to fulfill requirements or
```

File: spec/Section 2 -- Source Schema.md [hunk 5]
```diff
@@ -211,6 +211,85 @@ resolve entities and which source schemas merely contribute data to entities.
...
+
+type Query {
+  productById(id: ID!): Product
+  productBySku(sku: String!): Product @inaccessible
+}
+```
+
+The above example declares the key field `sku` to be inaccessible from
+the composite schema. However, type system members marked as inaccessible can
+still be used within the _execution schema_ to fulfill requirements or
```
Comment (michaelstaib) [inline spec/Section 2 -- Source Schema.md:243]:
```suggestion
still be used within the _execution schema_ to fulfil requirements or
```

File: spec/Section 2 -- Source Schema.md [hunk 6]
```diff
@@ -211,6 +211,85 @@ resolve entities and which source schemas merely contribute data to entities.
...
+  productBySku(sku: String!): Product @inaccessible
+}
+```
+
+The above example declares the key field `sku` to be inaccessible from
+the composite schema. However, type system members marked as inaccessible can
+still be used within the _execution schema_ to fulfil requirements or
+lookups.
+
+In contrast to the `@internal` directive the `@inaccessible` directive hides a
```
Comment (michaelstaib) [inline spec/Section 2 -- Source Schema.md:246]:
```suggestion
In contrast to the `@internal` directive the `@inaccessible` hides a
```

File: spec/Section 2 -- Source Schema.md [hunk 7]
```diff
@@ -211,6 +211,85 @@ resolve entities and which source schemas merely contribute data to entities.
...
+  productBySku(sku: String!): Product @inaccessible
+}
+```
+
+The above example declares the key field `sku` to be inaccessible from
+the composite schema. However, type system members marked as inaccessible can
+still be used within the _execution schema_ to fulfil requirements or
+lookups.
+
+In contrast to the `@internal` directive the `@inaccessible` hides a
```
Comment (michaelstaib) [inline spec/Section 2 -- Source Schema.md:246]:
```suggestion
In contrast to the `@internal` directive `@inaccessible` hides a
```

File: spec/Section 2 -- Source Schema.md [hunk 8]
```diff
@@ -211,6 +211,85 @@ resolve entities and which source schemas merely contribute data to entities.
...
+}
+```
+
+The above example declares the key field `sku` to be inaccessible from
+the composite schema. However, type system members marked as inaccessible can
+still be used within the _execution schema_ to fulfil requirements or
+lookups.
+
+In contrast to the `@internal` directive `@inaccessible` hides a
+type member from the composite schema even if other source schemas on the same type
```
Comment (michaelstaib) [inline spec/Section 2 -- Source Schema.md:247]:
```suggestion
type or output field from the composite schema even if other source schemas on the same type
```

File: spec/Section 2 -- Source Schema.md [hunk 9]
```diff
@@ -211,6 +211,85 @@ resolve entities and which source schemas merely contribute data to entities.
...
+```
+
+The above example declares the key field `sku` to be inaccessible from
+the composite schema. However, type system members marked as inaccessible can
+still be used within the _execution schema_ to fulfil requirements or
+lookups.
+
+In contrast to the `@internal` directive `@inaccessible` hides a
+type or output field from the composite schema even if other source schemas on the same type
+system member have no `@inaccessible` directive.
```
Comment (michaelstaib) [inline spec/Section 2 -- Source Schema.md:248]:
```suggestion
type or output field from the composite schema even if other source schemas on the
same type or output field have no `@inaccessible` directive.
```

**Pull request merged**